source/server/admin/prometheus_stats.cc

```diff
@@ -113,8 +110,11 @@ uint64_t outputStatType(
   }
 
   for (auto& group : groups) {
-    const std::string prefixed_tag_extracted_name =
-        PrometheusStatsFormatter::metricName(global_symbol_table.toString(group.first));
+    // We assume that custom metric flag is the same across all metrics with the same
+    // tagExtractedStatName. Callsites of defining these metrics must meet this criteria.
```

Comment:
agreed hard to enforce vs if we want to have ASSERT on this, it would add some computational const only for this in this code path. If not consistent, it's not that serious problem because it would just end up with or without envoy_ prefix on entire these metrics, and the output is in anyway consistent.
 
But note that in anyway is_custom_metric is activated only via C++ code, not by user-defined Wasm programs, so we will never encounter this inconsistency at runtime.

Reply:
What if someone writing WASM code creates a stat that has the same tag-extracted-name as a builtin?

WDYT of just allowing for that two stats may have the same tag-extracted-name but differing only in is_custom_metric. If nothing bad happens, can we remove this language about "must meet this criteria" and instead add unit tests for this case? I think it would only affect Prometheus also, and not anything else in the Envoy stats ecosystem.

This reminds me: should we flag is_custom_metric stats in the `/stats` endpoint in any way?